src/emulator/storage/apis/firebase.ts

```diff
@@ -561,12 +559,15 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
           });
         }
 
-        const md = finalizedUpload.file.metadata;
+        res.header("x-goog-upload-status", "final");
+        const uploadedFile = storageLayer.finalizeUpload(upload);
+
+        const md = uploadedFile.metadata;
         if (md.downloadTokens.length == 0) {
```

Comment:
nit: I know this is pre-existing code but please always use `===` when checking for equality. If you see other instances of `==` when changing functions, let's fix them as we go

 go/tsstyle#equality-checks

Reply:
Sure. But I'd prefer to make such fixes in a standalone PR. I can put one up after this.